src/main/java/org/cryptomator/cloudaccess/webdav/CachedPropfindEntryProvider.java

```diff
@@ -0,0 +1,230 @@
+package org.cryptomator.cloudaccess.webdav;
+
+import com.google.common.base.Preconditions;
+import org.cryptomator.cloudaccess.CachedNode;
+import org.cryptomator.cloudaccess.NodeCache;
+import org.cryptomator.cloudaccess.api.CloudPath;
+import org.cryptomator.cloudaccess.api.exceptions.CloudProviderException;
+import org.cryptomator.cloudaccess.api.exceptions.NotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.TimerTask;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+class CachedPropfindEntryProvider {
+
+	private static final Logger LOG = LoggerFactory.getLogger(CachedPropfindEntryProvider.class);
+
+	private final NodeCache cache;
+
+	private Function<CloudPath, PropfindEntryItemData> rootPoller;
+	private Function<CloudPath, List<PropfindEntryItemData>> cacheUpdater;
+
+	CachedPropfindEntryProvider(WebDavProviderConfig config, Function<CloudPath, PropfindEntryItemData> rootPoller, Function<CloudPath, List<PropfindEntryItemData>> cacheUpdater) {
+		this(new NodeCache());
+
+		this.rootPoller = rootPoller;
+		this.cacheUpdater = cacheUpdater;
+
+		var executor = Executors.newSingleThreadScheduledExecutor();
+		var remoteChangeDetector = new RemoteChangeDetector();
+		executor.scheduleWithFixedDelay(remoteChangeDetector, config.getRemoteChangePollerInitialDelaySeconds(), config.getRemoteChangePollerPeriodSeconds(), TimeUnit.SECONDS);
```

Comment:
despite the fact that no scheduler is required (we only need to expose the `pollRemoteChanges()` method), it is always good to keep a strong reference to an executor, so we can gracefully shut it down later.